### ISSUE TITLE
cordova@8.1.0 breaks when prompting for telemetry opt-in

### ISSUE THREAD

**Issue by rhysburnie** [apache/cordova-cli#337]
Hi, 

After installing `npm install -g cordova` no cordova commands work at all
Every command asks if you can track, answering yes or no make no difference
Ticks around a while then reports the following error
Tried reinstall, uninstall / reinstall
No luck

Cant check version because that too results in error - but I just ran normal install command

```
/Users/rgb/.nvm/versions/node/v8.9.4/lib/node_modules/cordova/node_modules/insight/lib/index.js:156
		prompt.close();
		       ^

TypeError: prompt.close is not a function
    at Insight.<anonymous> (/Users/rgb/.nvm/versions/node/v8.9.4/lib/node_modules/cordova/node_modules/insight/lib/index.js:156:10)
    at ontimeout (timers.js:475:11)
    at tryOnTimeout (timers.js:310:5)
    at Timer.listOnTimeout (timers.js:270:5)
```

Also tried globally installing an npm module called `prompt` incase that was the issue - no luck

**Comment by rhysburnie** [comment]
Looking at that file:
```
var permissionTimeout;

        var prompt = inquirer.prompt({
                type: 'confirm',
                name: 'optIn',
                message: msg || defaultMsg,
                default: true
        }, function (result) {
                // clear the permission timeout upon getting an answer
                clearTimeout(permissionTimeout);

                this.optOut = !result.optIn;
                cb(null, result.optIn);
        }.bind(this));

        // add a 30 sec timeout before giving up on getting an answer
        permissionTimeout = setTimeout(function () {
                // stop listening for stdin
                prompt.close();
```

Two things:
1. I gave answer well within 30sec timeout so why would it even call that
2. Either inquirer.prompt does not have .close() or there should be one added to the returned object?

**Comment by rhysburnie** [comment]
OK FYI the bug is already fixed in the latest on `insight` package - so you just need to update version

The current `insight` version being used in `cordova 8.1.0` uses a callback mechanism for the `inquirer` package - that package now has a promise based interface.

**Comment by brodybits** [comment]
Can you please share the following information:
- which platform
- output of `node --version`
- output of `npm --version`

I just tried Cordova CLI 8.1.0 on Node.js 4, 6, 8, and 10 on my Windows PC. I found issue on Node.js 4 and raised issue #339; `cordova --version` on Cordova CLI 8.1.0 works fine for me on Node.js 6, 8, and 10.

Possibly related: #338

**Comment by shazron** [comment]
I found a repro for this bug on node 8.12.0, npm 6.4.1

**Comment by raphinesse** [comment]
I guess the important part here is that you may not have a saved opt-in for this issue to surface.

@brodybits In hindsight, we should have tested the use-case that actually uses `inquirer` after fiddling with the transitive dependencies. I guess a release with older `inquirer` should work.

**Comment by raphinesse** [comment]
As I said before, this error only occurs when you have not answered the telemetry prompt before. Thus, a valid work-around would be to add `"optOut": false` or `"optOut": true` to `~/.config/configstore/insight-cordova.json`. I don't know where the file is located on Windows, unfortunately.

**Comment by brodybits** [comment]
I just reproduced this issue on Windows:

```sh
C:\Users\brodybits\Documents
λ rm -rf \Users\brodybits\.config\

C:\Users\brodybits\Documents
λ cordova --version
? May Cordova anonymously report usage statistics to improve the tool over time? Yes
C:\Users\brodybits\AppData\Local\nvs\node\10.11.0\x64\node_modules\cordova\node_modules\insight\lib\index.js:156
                prompt.close();
                       ^

TypeError: prompt.close is not a function
    at Insight.<anonymous> (C:\Users\brodybits\AppData\Local\nvs\node\10.11.0\x64\node_modules\cordova\node_modules\insight\lib\index.js:156:10)
    at ontimeout (timers.js:425:11)
    at tryOnTimeout (timers.js:289:5)
    at listOnTimeout (timers.js:252:5)
    at Timer.processTimers (timers.js:212:10)
```

**Comment by brodybits** [comment]
We just published Cordova CLI 8.1.1 which resolves this issue. In case you encounter this or any other issue in the future please do not hesitate to report it.
